pkg/virtctl/vnc/vnc.go

```diff
@@ -254,13 +283,29 @@ func checkAndRunVNCViewer(ctx context.Context, errChan chan error, port int) {
 			vncBin = TIGER_VNC
 			args = tigerVncArgs(port)
 		} else {
-			errChan <- fmt.Errorf("could not find %s or %s binary in $PATH",
-				REMOTE_VIEWER, TIGER_VNC)
-			errChan <- err
-			return
+			return "", nil, fmt.Errorf("could not find %s or %s binary in $PATH", REMOTE_VIEWER, TIGER_VNC)
 		}
 	default:
-		errChan <- fmt.Errorf("virtctl does not support VNC on %v", osType)
+		return "", nil, fmt.Errorf("virtctl does not support VNC on %v", osType)
+	}
+	return vncBin, args, nil
+}
+
+func checkAndRunVNCViewer(ctx context.Context, errChan chan error, port int) {
+	var err error
+	args := []string{}
+
+	vncBin := ""
+	osType := runtime.GOOS
```

Comment:
Inline `runtime.GOOS` into the two function calls

Reply:
It's used in the function calls, error handling and logging. Using `osType` makes it less error prone and easily to change (if the source for the OS type ever changes for some reason)

Reply:
If we chose to introduce unit tests for vnc then keeping the arg in `getUserSpecifiedVnc` and `getAutoDetectedVnc` makes sense. But this was more of a nit hinting at inlining this into the function calls in L294 and L296.

e.g. 

```
getUserSpecifiedVnc(ctx, runtime.GOOS, vncType, vncPath, port)
getAutoDetectedVnc(runtime.GOOS, port)
```

Reply:
While we're on the subject, virtctl vnc does not have any unit test right now. But would you mind adding some for `getUserSpecifiedVnc` and `getAutoDetectedVnc` if you think it's possible?

Reply:
Can I tackle VNC testing in another PR?